### PR TITLE
Updates conformance comparison report and exception handling

### PR DESCRIPTION
**Pull request by johnedquinn**
## Description
- Simplifies the UI of the conformance report. Here's the [latest report](https://github.com/partiql/partiql-lang-kotlin/actions/runs/10392893305). Here's an example of an [older report](https://github.com/partiql/partiql-lang-kotlin/actions/runs/10375182864).
  - I've made the sections easier to distinguish.
  - Enumerated the test cases that are failing/passing.
  - Instead of using green/red emojis for the column names of the table. I'm using them to indicate whether the actual results are good/bad.
  - Used better names for the reports.
  - Used the short hash when possible (while still showing the full hash in the details section).
- Simplifies the actual logic of report generation
- Updates the exception handling

Example of how the new green/red emojis are used to denote good/bad instead of using them as labels. It should make it easier for reviewers to read. Here's how it used to be:
| | Base (legacy) | eval | +/- |
| --- | ---: | ---: | ---: |
| % Passing | 90.72% | 96.05% | 5.33% |
| :white_check_mark: Passing | 5329 | 5643 | 314 |
| :x: Failing | 545 | 232 | -313 |
| :large_orange_diamond: Ignored | 0 | 0 | 0 |
| Total Tests | 5874 | 5875 | 1 |

And, here's how it is now:

| | BASE (LEGACY-B94D328) | TARGET (EVAL-B94D328) | +/- |
| --- | ---: | ---: | ---: |
| % Passing | 90.74% | 96.05% | 5.31% :white_check_mark: |
| Passing | 5330 | 5643 | 313 :white_check_mark: |
| Failing | 544 | 232 | -312 :white_check_mark: |
| Ignored | 0 | 0 | 0 :white_check_mark: |
| Total Tests | 5874 | 5875 | 1 :white_check_mark: |

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ReportAnalyzer.kt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        val delta = second - first
+        val deltaIcon = getIconForComparison(first, second, positiveDeltaGood)
+        return "| $name | $first | $second | $delta $deltaIcon |"
+    }
+
+    private fun <T> getIconForComparison(first: Comparable<T>, second: T, positiveDeltaGood: Boolean): String {
+        val comparison = first.compareTo(second)
+        return when {
+            comparison < 0 -> if (positiveDeltaGood) ICON_CHECK else ICON_CIRCLE_RED
+            comparison == 0 -> ICON_CHECK
```
Comment (alancai98) [inline test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ReportAnalyzer.kt:89]: nit: wonder if for if there's no diff, we want some neutral indication consider :heavy_minus_sign: (`:heavy_minus_sign:`). ✅ would seem to indicate some good change but perhaps no regression is good?

**Review by alancai98 — APPROVED** [review]
nice work on the refactor of both the comparison report and test runner exception handling. only left a few minor comments/suggestions

File: test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ReportAnalyzer.kt (hunk 1 above)
Comment (johnedquinn) [reply test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ReportAnalyzer.kt:89]: Yeah, I played with doing a grey icon, but it started to look a bit cluttered visually. I figured green/red are pretty easy indicators for reviewers that something is good/bad.

File: test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/test/TestRunner.kt [hunk 2]
```diff
@@ -26,32 +26,39 @@ class TestRunner<T, V>(private val factory: TestExecutor.Factory<T, V>) {
...
-                is Assertion.EvaluationSuccess -> throw IllegalStateException("Expected success but exception thrown.", e)
-                is Assertion.EvaluationFailure -> {} // skip
+            is Assertion.EvaluationFailure -> {
+                var thrown: Throwable? = null
+                val ion = try {
+                    val statement = executor.prepare(input)
+                    val actual = executor.execute(statement)
+                    executor.toIon(actual)
+                } catch (t: Throwable) {
+                    thrown = t
```
[the remark this answers is not in the document]
Comment (johnedquinn) [reply test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/test/TestRunner.kt:53]: True! While it's not part of this PR, I'll give some insight.

There's actually a reason for the lack of a stack trace -- performance. From my performance evaluation, one thing was clear: `fillInStackTrace` is insanely expensive on the CPU. The `TypeCheckException` removes the default implementation of `fillInStackTrace`. If you'd like more information, you can temporarily comment it out to figure out where it's coming from. Beyond that, you're totally right. More of the `TypeCheckExceptions` should have a corresponding error message -- though not all instances have them.

**Pull request merged**